prestoadmin/yarn_slider/server.py

```diff
@@ -0,0 +1,150 @@
+
+from prestoadmin.yarn_slider.config import SliderConfig, \
+    DIR, SLIDER_USER, APPNAME, JAVA_HOME, HADOOP_CONF, SLIDER_MASTER, \
+    PRESTO_PACKAGE, SLIDER_CONFIG_DIR
+from prestoadmin.util.base_config import requires_config
+
+from prestoadmin.util.fabricapi import task_by_rolename
+
+__all__ = ['slider_install', 'slider_uninstall', 'install', 'uninstall']
+
+
+SLIDER_PKG_DEFAULT_FILES = ['appConfig-default.json', 'resources-default.json']
+
+
+@task
+@requires_config(SliderConfig)
+@task_by_rolename(SLIDER_MASTER)
+def slider_install(slider_tarball):
+    """
+    Install slider on the slider master. You must provide a tar file on the
+    local machine that contains the slider distribution.
+
+    :param slider_tarball: The gzipped tar file containing the Apache Slider
+    distribution
+    """
+    deploy_install(slider_tarball)
+
+
+def deploy_install(slider_tarball):
+    slider_dir = env.conf[DIR]
+    slider_parent = os.path.dirname(slider_dir)
+    slider_file = os.path.join(slider_parent, os.path.basename(slider_tarball))
+
+    sudo('mkdir -p %s' % (slider_dir))
+
+    result = put(slider_tarball, os.path.join(slider_parent, slider_file))
+    if result.failed:
+        abort('Failed to send slider tarball %s to directory %s on host %s' %
+              (slider_tarball, slider_dir, env.host))
+
+    sudo('gunzip -c %s | tar -x -C %s --strip-components=1 && rm -f %s' %
+         (slider_file, slider_dir, slider_file))
+
+
+@task
+@requires_config(SliderConfig)
+@task_by_rolename(SLIDER_MASTER)
+def slider_uninstall():
+    """
+    Uninstall slider from the slider master.
+    """
+    sudo('rm -r "%s"' % (env.conf[DIR]))
+
+
+def get_slider_bin(conf):
+    return os.path.join(conf[DIR], 'bin', 'slider')
+
+
+def run_slider(slider_command, conf):
+    with shell_env(JAVA_HOME=conf[JAVA_HOME],
+                   HADOOP_CONF_DIR=conf[HADOOP_CONF]):
+        return sudo(slider_command, user=conf[SLIDER_USER])
+
+
+@task
+@requires_config(SliderConfig)
+@task_by_rolename(SLIDER_MASTER)
+def install(presto_yarn_package):
+    """
+    Install the presto-yarn package on the cluster using Apache Slider. The
+    presto-yarn package takes the form of a zip file that conforms to Slider's
+    packaging requirements. After installing the presto-yarn package the presto
+    application is registered with Slider.
+
+    The name of the presto application is arbitrary and set in the slider
+    configuration file. The default is PRESTO
+
+    :param presto_yarn_package: The zip file containing the presto-yarn
+    package as structured for Slider.
+    """
+    conf = env.conf
+    package_filename = os.path.basename(presto_yarn_package)
+    package_file = os.path.join('/tmp', package_filename)
+
+    result = put(presto_yarn_package, package_file)
+    if result.failed:
+        abort('Failed to send slider application package to %s on host %s' %
+              (package_file, env.host))
+
+    package_install_command = \
```

Comment:
where are you creating the hdfs dir /user/yarn/ and chown it by user yarn ? the package --install would fail without it? https://github.com/prestodb/presto-yarn/blob/master/README.md#preparing-other-slider-specific-configuration has the steps we do manually. looks like hdfs dir creation is needed prior to install?